metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/loader/xml/SraXmlParser.java

```diff
@@ -15,16 +15,18 @@
  * limitations under the License.
  *
  */
-package uk.ac.ebi.ampt2d.metadata.loader.core.xml;
+package uk.ac.ebi.ampt2d.metadata.loader.xml;
 
 import org.apache.xmlbeans.XmlException;
 
 public abstract class SraXmlParser<SRA_OBJECT> {
 
-    String XML_ROOT_TAGS = "(</ROOT>|<ROOT.*display=xml\">)";
+    private final String XML_ROOT_TAGS = "(</ROOT>|<ROOT.*display=xml\">)";
 
     public abstract SRA_OBJECT parseXml(String xmlString, String accession) throws XmlException;
 
+    protected abstract String removeSetTagsFromXmlString(String xmlString);
```

Comment:
Maybe the implementation for this method could also be provided here to keep it consistent with `removeRootTagsFromXmlString` method. But as I mentioned above, the logic for both methods could be moved to the retriever classes.

Reply:
The problem is that the set tags are different for each object type :cry: (ANALYSIS_SET, PROJECT_SET, etc). We could use a more generic regex but I suspect that could backfire at some point.

Reply:
If you look at the root element from the API response, it is using regex to remove. I just checked SRA schema and they all start with `OBJECT_SET` except for `DATASET` which is `DATASETS`.